### PR TITLE
New package for general-purpose zero-knowledge circuits

### LINKED ISSUE #73: Create a package for Circom circuits
**Tasks**:

- [ ] Create `@zk-kit/circuits` package
- [ ] Move `lazy-tower.circom` to package
- [ ] Create `lean-imt.circom`
- [ ] Create `poseidon-proof.circom`

### PR DESCRIPTION
**Pull request by cedoor**
## Description

A new package for providing general-purpose zero-knowledge circuits has been added. As other projects/protocols become modularized, reusable circuits could be managed in this package, tested, and documented.

`@zk-kit/circuits` can be installed with NPM (or Yarn) and its circuits accessed from `node_modules/@zk-kit/circuits`.

## Related Issue

Closes #73 

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: packages/circuits/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+</div>
+
+
+| This package offers a collection of reusable circuits designed for integration into other projects or protocols, promoting code modularization within the zero-knowledge ecosystem. |
+| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+
+## Circuits
+
+* Circom:
+    * [PoseidonProof](./circom/poseidon-proof.circom): It proves the possession of a Posidon pre-image without revealing the pre-image itself.
```
Comment (0xjei) [inline packages/circuits/README.md:38]: nit: type `Posidon´ to `Poseidon`

File: packages/circuits/noir/.gitkeep
Comment (0xjei) [inline packages/circuits/noir/.gitkeep:1]: Q: Why do we want to keep this folder for `Noir`? Do we also want to add implementations in that language, or is it autogenerated by `circom-kit`?

**Review by 0xjei — COMMENTED** [review]
Partial-review: great configs but the npm scripts are not working right now, left some nits! Great work so far :+1:

File: packages/circuits/noir/.gitkeep
Comment (cedoor) [reply packages/circuits/noir/.gitkeep:1]: @fabianschu is going to add some Noir circuits :) -> https://github.com/privacy-scaling-explorations/zk-kit/issues/73#issuecomment-1830295070

**Comment by cedoor** [comment]
> ```shell
> yarn circom:compile poseidon-proof
> 
> ===| Compiling the circuit |===
> Error: Compiler error:
> Error: Command failed: circom --sym --wasm --r1cs -p bn128 -o ./build/poseidon-proof -l ../../node_modules/circomlib/circuits --verbose --inspect --O1 ./circuits/main/poseidon-proof.circom
> error[P1014]:  The file ../poseidon-proof.circom to be included has not been found
>  = Consider using compilation option -l to indicate include paths
> ```

It's a `circomkit` bug: https://github.com/erhant/circomkit/issues/33. I'll update it as soon as it's solved.

**Review by 0xjei — APPROVED** [review]
Great! This is going to be a masterpiece for circuit development not only for PSE. Just left some comments and nits but ready to be merged :)

**Pull request merged**